src/main/kotlin/anttracker/Release.kt

```diff
@@ -23,31 +147,76 @@ value class ReleaseId(
-}
-
-class Release(
-    val id: ReleaseId,
-    val product: Product,
-    var releaseDate: LocalDate,
-)
 
-// -------------------------------------------------------------------------------
-// Prints a sub-menu of selecting a product to create a release for.
-// Use by including in main-menu loop; should be triggered upon user selecting
-// New Release from the main menu. Thus, upon function return, system returns to main menu.
-// ---
-fun menu() {
-    TODO()
+    override fun toString(): String = id
 }
 
 // -------------------------------------------------------------------------------
-// Prints to console a paginated list of releases for a product
-// Returns the release selected by the user, or null if the user changed their mind.
-// Call as part of any relevant sub-menu
-//  e.g. During request creation, call this to find and select the affected release
+// Implementation of a PageOf Class as PageOfReleases
+// Each PageOf class needs to define:
+//      - init{} block, which MUST call initLastPageNum(). See below why this is necessary.
+//      - display(), to define how the page is displayed to console
+//      - queryToDB(), to define the DAO query to DB used to pull records into memory
 // ---
-fun selectRelease(
-    product: Product, // in
-): Release? {
-    TODO()
+class PageOfReleases(
+    private val productName: String,
+) : PageOf<Release>() {
+    // -------------------------------------------------------------------------------
+    // Automatically called upon object creation (i.e. a 2nd ctor)
+    // Not defined in PageOf superclass as that gets called BEFORE subclass properties
+    //      (such as productName) are defined. This can result in the calculation for
+    //      lastPageNum being undefined behaviour as the query isn't fully-formed.
+    //      To prevent this, the init block is (annoyingly) defined in the subclass
+    //      which ensures it is called AFTER the constructor in the class header.
+    // ---
+    init {
+        initLastPageNum()
+    }
+
+    // -------------------------------------------------------------------------------
+    // Prints a single product release record to console.
+    // Should look like:
+    // x.x.x.x.    YYYY/MM/DD
+    // ---
+    override fun printRecord(
+        record: Release, // in
+    ) {
+        val strRelId: String = record.releaseId
+        val strRelDate: String = record.releaseDate.format(DateTimeFormatter.ofPattern("yyyy/MM/dd"))
+        println(strRelId.padEnd(12) + strRelDate)
+    }
+
+    // -------------------------------------------------------------------------------
+    // In SQL this would be:
+    //  SELECT * FROM release
+    //  WHERE product = productName
+    //  ORDER BY product, releaseDate DESC
+    //  LIMIT qLimit OFFSET pagenum * qLimit + qOffset
+    // ---
+    override fun queryToDB(): SizedIterable<Release>? {
+        var output: SizedIterable<Release>? = null
+        val test = productName
+        transaction {
```

Comment:
I think we can do this in one query. 

```kotlin
 (Releases innerJoin Products)
                .select(Releases.columns)
                .where { Products.name eq productName }
                .limit(n = qLimit, offset = (pagenum * qLimit + qOffset).toLong())
                .orderBy(
                    Releases.product to SortOrder.DESC,
                    Releases.releaseDate to SortOrder.DESC,
                )
                .map { Release.wrapRow(it) }
```

Reply:
I was trying that approach but couldn't quite get it to work. That does look good so I will see about implementing it.